syntax = "proto3";

package account.v1;

// import "common/v1/common.proto"; // 导入公共定义（分页、错误等）

service AccountService{
    rpc ListEmailAddresses(ListEmailAddressesReq) returns (ListEmailAddressesResp);     // 列出所有可选的邮箱地址（邮箱类别的账号）

    rpc CreateAccount(CreateAccountReq) returns (CreateAccountResp);        // 创建账号
    rpc ListAccountById(ListAccountByIdReq) returns (ListAccountByIdResp);  // 根据 session_id 和 account_id 查询账号
    rpc ListAccounts(ListAccountsReq) returns (ListAccountsResp);           // 根据 session_id 查询账号
    rpc UpdateAccount(UpdateAccountReq) returns (UpdateAccountResp);        // 更新账号
    rpc DeleteAccount(DeleteAccountReq) returns (DeleteAccountResp);        // 删除账号

    rpc FetchPassword(FetchPasswordReq) returns (FetchPasswordResp);        // 获取密码
    rpc ChangePassword(ChangePasswordReq) returns (ChangePasswordResp);     // 仅修改密码
}

message ViewAccount{
    string account_id = 1;
    string provider_name = 2;
    string platform_name = 3;
    string username = 4;
    string nickname = 5;
    string sub_account = 6;
    string mobile_phone = 7;
    string email = 8;
    string postscript = 9;
    string url = 10;
    string hotline = 11;
    string category = 12;
    string created_time = 13;
    string updated_time = 14;
}

message Account{
    string provider_name = 1;
    string platform_name = 2;
    string username = 3;
    string nickname = 4;
    string sub_account = 5;
    string mobile_phone = 6;
    string email = 7;
    string postscript = 8;
    string url = 9;
    string hotline = 10;
    string category = 11;
}

message ListEmailAddressesReq{
    string session_id = 1;
}

message ListEmailAddressesResp{
    repeated string email_addresses = 1;
    string message = 2;
}

message CreateAccountReq{
    string session_id = 1;
    Account account = 2;
    string password = 3;
    string main_key = 4;
}

message CreateAccountResp{
    bool result = 1;
    string message = 2;
}

message ListAccountByIdReq{
    string session_id = 1;
    string account_id = 2;
}

message ListAccountByIdResp{
    ViewAccount account = 1;
    string message = 2;
}

/*
message ListAccountsReq{
    string session_id = 1;
    common.v1.Filter filter = 2;
    int32 page_size = 3;
}

message ListAccountsResp{
    repeated Account accounts = 1;
    string message = 2;
}*/

message ListAccountsReq{
    string session_id = 1;
    Account conditions = 2;
    int32 page_size = 3;
}

message ListAccountsResp{
    repeated ViewAccount accounts = 1;
    string message = 2;
}

message UpdateAccountReq{
    string session_id = 1;
    string account_id = 2;
    Account account = 3;
    string password = 4;
    string main_key = 5;
}

message UpdateAccountResp{
    bool result = 1;
    string message = 2;
}

message DeleteAccountReq{
    string session_id = 1;
    string account_id = 2;
    string main_key = 3;
}

message DeleteAccountResp{
    bool result = 1;
    string message = 2;
}

message FetchPasswordReq{
    string session_id = 1;
    string account_id = 2;
    string main_key = 3;
}

message FetchPasswordResp{
    string password = 1;
    string message = 2;
}

message ChangePasswordReq{
    string session_id = 1;
    string account_id = 2;
    string main_key = 3;
    string password = 4;
}

message ChangePasswordResp{
    bool result = 1;
    string message = 2;
}
